frontend/src/pages/admin/NotificationsHub.vue

```diff
@@ -50,78 +82,133 @@
+            billingStates: [
+                'Active',
+                'Trial',
+                'Unmanaged',
+                'Canceled'
+            ],
             errors: {
 
             }
         }
     },
     computed: {
+        ...mapState('account', ['features']),
         roleIds () {
             return Object.values(RoleNames).filter(r => r !== 'none').reverse().map(r => r[0].toUpperCase() + r.substring(1))
         },
         canSubmit () {
             return this.form.title.length > 0 &&
                 this.form.message.length > 0 &&
-                this.form.roles.length > 0
+                this.form.roles.length > 0 &&
+                this.form.teamTypes.length > 0 &&
+                (!this.features.billing || this.form.billing.length > 0)
         },
         urlPlaceholder () {
             return this.form.externalUrl ? 'https://flowfuse.com' : '{ name: "<component-name>", params: {id: "<id>"} }'
         }
     },
+    async created () {
+        const teamTypes = (await teamTypesApi.getTeamTypes(null, null, 'all')).types
+        this.teamTypes = teamTypes.map(tt => {
+            return {
+                order: tt.order,
+                id: tt.id,
+                name: tt.name,
+                active: tt.active
+            }
+        })
+        this.teamTypes.sort((A, B) => {
+            if (A.active === B.active) {
+                return A.order - B.order
+            } else if (A.active) {
+                return -1
+            } else {
+                return 1
+            }
+        })
+    },
     methods: {
         getAnnouncements () {
             return adminApi.getAnnouncementNotifications()
                 .then(res => console.info(res))
         },
         submitForm () {
             return this.sendAnnouncementNotification({ mock: true })
-                .then(mockRes => Dialog.show({
-                    header: 'Platform Wide Announcement',
-                    kind: 'danger',
-                    text: `You are about to send an announcement to ${mockRes.recipientCount} recipients.`,
-                    confirmLabel: 'Continue'
-                }, async () => this.sendAnnouncementNotification({ mock: false })))
+                .then(mockRes => {
+                    if (mockRes.recipientCount === 0) {
+                        Dialog.show({
+                            header: 'Platform Wide Announcement',
+                            text: 'Your filters matched no users.',
+                            confirmLabel: 'Cancel',
+                            canBeCanceled: false
+                        })
+                    } else {
+                        Dialog.show({
+                            header: 'Platform Wide Announcement',
+                            kind: 'danger',
+                            text: `You are about to send an announcement to ${mockRes.recipientCount} ${pluralize('user', mockRes.recipientCount)}.`,
+                            confirmLabel: 'Continue'
```

Comment:
```suggestion
                            confirmLabel: 'Continue',
                            canBeCanceled: false
```

attempting to send notifications to 0 users, refining parameters and sending a mock response caused the dialog to pop up without the possibility of canceling (i had to refresh the page)

Reply:
Odd - sounds like a bug with the Dialog service if a previous call can influence a future call.

Reply:
Ooops - this should be set to `true` not `false`

Reply:
my bad!